### ISSUE TITLE
a few tune-ups for the PDF contribution reports

### ISSUE THREAD

**Issue by mrose17** [brave/browser-laptop#6039]
**Did you search for similar issues before submitting this one?** yes

**Describe the issue you encountered:** 
1. Verified publishers not shown as verified
2. The OK button is not placed properly (see attachment)
3. the PDF file is generated in a new tab which then disappears and a file save dialog comes up. this is jarring. wouldn't it be better to open a new tab, generate the PDF and then let the user view it normally, and then let them decide to save it?

**Expected behavior:**
1. show verified publishers with icon
2. place OK button properly
3. when a contribution report is generated, just open a new tab and show it as a PDF

- Platform (Win7, 8, 10? macOS? Linux distro?): macOS

- Brave Version: 0.12.10 (master)

- Steps to reproduce:
    1. go to about:preferences#payments
    2. click on 'View Payment History"
    3. click on a contribution

- Screenshot if needed:
![0](https://cloud.githubusercontent.com/assets/400552/20911266/b96d7c98-bb1b-11e6-897a-58ad573ef9c9.png)

- Any related issues:
none.
![0](https://cloud.githubusercontent.com/assets/400552/20911263/b611d9e0-bb1b-11e6-9d9d-cf131224f33e.png)

**Comment by luixxiul** [comment]
I'd like to take the button alignment issue.

**Comment by luixxiul** [comment]
@mrose17 @willy-b 
> 1. show verified publishers with icon

I addressed the issue with https://github.com/brave/browser-laptop/pull/6009/commits/21ece29eab80de4af2c606a65b54c13764b04c93 based on @bsclifton 's review https://github.com/brave/browser-laptop/pull/6009#issuecomment-264996591

> 2. place OK button properly

I will address this issue within the #6009

**Comment by mrose17** [comment]
@luixxiul - thanks!

**Comment by willy-b** [comment]
Hey @luixxiul, can we split the Payment History modal OK button issue from the Contribution Statement PDF bugs @mrose17 brought up here? That OK button regression is entirely separate and so I think if I fix the PDF Statement issues we won't step on each other's toes at all.

**Comment by luixxiul** [comment]
Agree, will do! @willy-b

**Comment by willy-b** [comment]
That seems like yet another issue:
 #6059 is about Payment History having correct past/future context, not the OK button position, and the screenshot is of a different version of Brave which is before the Contribution PDF was added.

Can the OK button be it's own issue? It doesn't exist in the version in the screenshot from @bradleyrichter in #6059.

**Comment by luixxiul** [comment]
@willy-b my bad, you're right!

**Comment by bsclifton** [comment]
great- those two other issues are now split out; let us know if there's anything else you need, @willy-b :smile: thanks for reviewing!

**Comment by mrose17** [comment]
@bsclifton - can we close this issue now?

**Comment by willy-b** [comment]
@mrose17, don't think so. I'm working on this now. To my knowledge no one has pushed a fix for verified publishers or opening the generated contribution PDF in PDF.js instead of launching a download.

**Comment by mrose17** [comment]
@willy-b - thanks. do you think you can address this issue?

**Comment by willy-b** [comment]
yes, I'll let you know if I encounter problems.
working on verified publishers right now, will update here when that is finished and I begin work on opening with PDF.js.

do you know what release this is targeted for?

**Comment by mrose17** [comment]
i'm hoping end of next week? is that reasonable?

**Comment by willy-b** [comment]
sounds good, go ahead and set a milestone if you like

**Comment by mrose17** [comment]
will do, thanks!

**Comment by willy-b** [comment]
I pushed the verified publisher icon fix last week and the open-in-PDF.js fix is ready, though it depends on being able to open local PDFs which may have just been fixed as #6330 cc / @bbondy (previously was a bug as #2714)

**Comment by willy-b** [comment]
Just opened a PR for the one-liner verified publisher icon fix (or feel free to make change yourself as part of one of your commits, you'll see it's really minor).

**Comment by willy-b** [comment]
Hey @mrose17 , update: 
The publisher icon fix is ready in #6375, but the PDF.js fix will not be ready today as changes in PDF.js have to be made and I am not familiar enough.

Also the issue for the original feature has been reopened (#4769) due to a regression on Windows and possibly elsewhere, so I need to work on that.

**Comment by mrose17** [comment]
@willy-b thanks for the update... many thanks for sticking with this!

cc: @bsclifton

**Comment by willy-b** [comment]
hey @mrose17, another status update:

The situation with the underlying feature (contrib stmt PDFs) being broken during Chromium54 upgrade is now mostly figured out (#4769 regression).

The remaining webview API regression that this depends on is `webview.printToPDF`.
It was broken during the Chromium54 upgrade (being fixed in #6450) and is needed (even with PDF.js change) because it generates the actual PDF doc in first place.

In chasing the printToPDF issue (@bbondy assigned himself last night in #6450 but I have been looking in parallel) I have been catching up on Electron/Muon standalone build being replaced by browser-laptop-bootstrap in last 2 weeks (https://github.com/brave/muon/issues/132 , PR https://github.com/brave/muon/pull/136). Still getting the new browser-laptop-bootstrap build going, issues I have with that are/will be tracked in https://github.com/brave/browser-laptop-bootstrap/issues/9.

thanks.

**Comment by willy-b** [comment]
@mrose17, quick clarification:
by tomorrow you are expecting fixes for the printToPDF (#6450) and setTabIndex (#6442) muon changes that broke the original feature?

or two items you requested here?
 - the verified publisher fix (done, #6375) 
 - opening the PDF in PDF.js after generating it (depends on #2714)

given the original feature is still pending #6450, #6442 you can have #6375 now but PDF.js will not be here tomorrow :-/

**Comment by mrose17** [comment]
my expectation is whatever your comfortable with!

**Comment by bbondy** [comment]
print-to-pdf is working on mac and linux on this branch: print-to-pdf
It isn't tested yet on Windows, but it might need another fix there.

The setTabIndex stuff should be resolved now with Preview7 which you'll get if you npm install now. (Electron 2.0.7

**Comment by willy-b** [comment]
The print-to-pdf branch in `brave/muon`, not browser-laptop (this repo), right?

when will it hit browser-laptop?

**Comment by willy-b** [comment]
Fix in Muon doesn't mean I or many others can test it or work on it: 
Muon can't be built standalone anymore (https://github.com/brave/muon/issues/132) and the setup/build for browser-laptop-bootstrap is slow and tricky (e.g. https://github.com/brave/browser-laptop-bootstrap/issues/9 )

**Comment by mrose17** [comment]
@bsclifton - would it be OK to ask @willy-b to submit a PR to get this into https://github.com/brave/browser-laptop/tree/0.13.1-branch ?

**Comment by bsclifton** [comment]
@mrose17 PRs definitely welcome :smile: 

Since this issue is kind of a "catch all", I'd suggest breaking out the items being fixed into discrete issues which are easy to test (and capture all the info about the issue). I'm not sure which issue his PR would be addressing?

**Comment by mrose17** [comment]
@willy-b - can you generate a PR ?

@bsclifton - the three things in the issue are kind of small. do we really need to split this issue into three?

**Comment by bsclifton** [comment]
@mrose17 After reading through the original post, it should be OK (sorry I had just glanced before) :smile:

**Comment by mrose17** [comment]
no worries!

**Comment by willy-b** [comment]
there's a PR #6375 for the publisher icon.
the other part (opening in new tab) depends on #2714 (ability of Brave to open local PDFs) which is still outstanding.

**Comment by mrose17** [comment]
@willy-b - thanks! i'll see what we can do...

**Comment by bsclifton** [comment]
Fixed with https://github.com/brave/browser-laptop/pull/6375
